src/views/gitStashExplorer.ts

```diff
@@ -0,0 +1,43 @@
+'use strict';
+import { Event, EventEmitter, ExtensionContext, TreeDataProvider, TreeItem, Uri, window } from 'vscode';
+import { ExplorerNode, StashNode } from './gitExplorerNodes';
+import { GitService, GitUri } from '../gitService';
+import { StashCommitNode } from './stashCommitNode';
+
+export * from './gitExplorerNodes';
+
+export class GitStashExplorer implements TreeDataProvider<ExplorerNode>  {
+    private _node: ExplorerNode;
+    private _onDidChangeTreeData = new EventEmitter<ExplorerNode>();
+    public get onDidChangeTreeData(): Event<StashCommitNode> {
+        return this._onDidChangeTreeData.event;
+    }
+
+    constructor(private context: ExtensionContext, private git: GitService) {
+        const editor = window.activeTextEditor;
+
+        const uri = (editor !== undefined && editor.document !== undefined)
+            ? new GitUri(editor.document.uri, { repoPath: git.repoPath, fileName: editor.document.uri.fsPath })
+            : new GitUri(Uri.file(git.repoPath), { repoPath: git.repoPath, fileName: git.repoPath });
+
+        this._node = new StashNode(uri, this.context, this.git);
```

Comment:
What do you think of keeping this in a node or just nuking the node and dropping the logic in the explorer? I keep going back and forth -- eh keeping it in the node is nice for possible reuse later in another view or whatnot